src/index.css

```diff
@@ -4,6 +4,21 @@
   text-rendering: optimizelegibility;
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
+  color-scheme: dark light;
+  background-color: var(--background-color);
+  color: var(--text-color);
+  --cart-icon-color: #111;
+}
+
+:root.dark {
```

Comment:
check your logic because when i choose dark theme, text color dark too and check header color.

1.  
<img width="925" alt="Screenshot 2024-05-21 at 19 16 27" src="https://github.com/3apy6a/fe-react-2024/assets/115080771/952ddaac-4419-4c10-88a2-e39947945a41">

 2.
<img width="719" alt="Screenshot 2024-05-21 at 19 16 09" src="https://github.com/3apy6a/fe-react-2024/assets/115080771/8a649b33-dabc-4d90-9151-94a37f6fe74f">

3.
<img width="1321" alt="Screenshot 2024-05-21 at 19 16 51" src="https://github.com/3apy6a/fe-react-2024/assets/115080771/c0273a45-0da5-4e15-8122-93b8862a1c32">